nncf/experimental/onnx/graph/model_transformer.py

```diff
@@ -11,118 +11,106 @@
  limitations under the License.
 """
 
-from typing import TypeVar
-
 from copy import deepcopy
-import onnx
+from typing import List, TypeVar
 
-from nncf.common.graph.model_transformer import ModelTransformer
+import onnx
+from nncf.common.graph.definitions import NNCFGraphNodeType
+from nncf.common.graph.graph import NNCFGraph
 from nncf.common.graph.transformations.commands import TargetType
 from nncf.common.quantization.structs import QuantizationMode
-from nncf.common.graph.definitions import NNCFGraphNodeType
 from nncf.common.utils.logger import logger as nncf_logger
-from nncf.experimental.onnx.graph.onnx_graph import ONNXGraph
 from nncf.experimental.onnx.graph.nncf_graph_builder import GraphConverter
-from nncf.experimental.onnx.graph.transformations.layout import ONNXTransformationLayout
-from nncf.experimental.onnx.graph.transformations.commands import ONNXQuantizerInsertionCommand
+from nncf.experimental.onnx.graph.onnx_graph import ONNXGraph
 from nncf.experimental.onnx.graph.transformations.commands import ONNXOutputInsertionCommand
+from nncf.experimental.onnx.graph.transformations.commands import ONNXQuantizerInsertionCommand
+from nncf.experimental.onnx.graph.transformations.layout import ONNXTransformationLayout
+from nncf.experimental.post_training.compression_builder import MODEL_TRANSFORMERS
+from nncf.experimental.post_training.graph.model_transformer import ModelTransformerMeta
 
 ModelType = TypeVar('ModelType')
 
 
 # pylint: disable=no-member
-
-class ONNXModelTransformer(ModelTransformer):
+@MODEL_TRANSFORMERS.register()
```

Comment:
Will there be many model transformers, do we need to dispatch model transformers immediately based on the model object type, or will the framework-specific model transformer be only first instantiated inside another framework-specific class? If latter, then we don't need `MODEL_TRANSFORMERS` as a separate registry. If former, then the registry should be rather called `PTQ_MODEL_TRANSFORMERS`, I think.

Reply:
This solution is directed to the future expanding the set of backends. Currently, the backend-specific model transformer calls in the compression builder. So, I think, that using a registry is one of the solutions to have one instance while the PTQ pipeline executes. In any case, I am open to better solutions :)

Reply:
Ok, then rename the registry and create a basic factory/dispatcher function (even if the only option that it will currently provide is the ONNX backend or failure). Having `MODEL_TRANSFORMERS.get()` everywhere is not extensible in the future, so lay the groundwork for future extension now.